### ISSUE TITLE
All proxies returning RequestError: Proxy responded with 400: 155 bytes

### ISSUE THREAD

**Issue by dhavdc** [apify/got-scraping#53]
```
RequestError: Proxy responded with 400: 155 bytes
    at Request._beforeError (/Users/daniel/Documents/test/node_modules/got-cjs/dist/source/core/index.js:325:21)
    at Request.flush (/Users/daniel/Documents/test/node_modules/got-cjs/dist/source/core/index.js:314:18)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at ClientRequest.<anonymous> (/Users/daniel/Documents/test/node_modules/got-scraping/dist/resolve-protocol.js:34:28)
    at Object.onceWrapper (node:events:514:26)
    at ClientRequest.emit (node:events:394:28)
    at Socket.socketOnData (node:_http_client:525:11)
    at Socket.emit (node:events:394:28)
    at addChunk (node:internal/streams/readable:315:12)
    at readableAddChunk (node:internal/streams/readable:289:9)
    at Socket.Readable.push (node:internal/streams/readable:228:10)
    at TCP.onStreamRead (node:internal/stream_base_commons:199:23) {
  input: undefined,
  code: 'ERR_GOT_REQUEST_ERROR',
  timings: undefined,

```
Proxy is in the format proxyUrl: 'http://172.64.75.188:80' and has been tested to work. I am not using any extra options besides proxyUrl and link.

**Comment by szmarczak** [comment]
Does it work with a url like `https://example.com`

**Comment by dhavdc** [comment]
> Does it work with a url like `https://example.com` ?

It worked with a localhost mitmweb proxy http://localhost:8080. I still can't get it to work with the proxies I need to use. No matter what link I use for the url it is the same response from the proxy. They are hosted on cloudflare and you can see it returns something in the rawheaders.

`{
  host: 'example.com',
  port: 443,
  rejectUnauthorized: false,
  ALPNProtocols: [ 'h2', 'http/1.1' ],
  servername: 'example.com',
  path: undefined
}
ClientRequest {
  _events: [Object: null prototype] {},
  _eventsCount: 0,
  _maxListeners: undefined,
  outputData: [],
  outputSize: 0,
  writable: true,
  destroyed: false,
  _last: true,
  chunkedEncoding: false,
  shouldKeepAlive: false,
  _defaultKeepAlive: true,
  useChunkedEncodingByDefault: false,
  sendDate: false,
  _removedConnection: false,
  _removedContLen: false,
  _removedTE: false,
  _contentLength: null,
  _hasBody: true,
  _trailer: '',
  finished: false,
  _headerSent: false,
  _closed: false,
  socket: null,
  _header: null,
  _keepAliveTimeout: 0,
  _onPendingData: [Function: nop],
  agent: Agent {
    _events: [Object: null prototype] {
      free: [Function (anonymous)],
      newListener: [Function: maybeEnableKeylog]
    },
    _eventsCount: 2,
    _maxListeners: undefined,
    defaultPort: 80,
    protocol: 'http:',
    options: [Object: null prototype] { path: null },
    requests: [Object: null prototype] {},
    sockets: [Object: null prototype] { '172.67.182.29:80:': [Array] },
    freeSockets: [Object: null prototype] {},
    keepAliveMsecs: 1000,
    keepAlive: false,
    maxSockets: Infinity,
    maxFreeSockets: 256,
    scheduling: 'lifo',
    maxTotalSockets: Infinity,
    totalSocketCount: 1,
    [Symbol(kCapture)]: false
  },
  socketPath: undefined,
  method: 'CONNECT',
  maxHeaderSize: undefined,
  insecureHTTPParser: undefined,
  path: 'example.com:443',
  _ended: false,
  res: null,
  aborted: false,
  timeoutCb: null,
  upgradeOrConnect: false,
  parser: null,
  maxHeadersCount: null,
  reusedSocket: false,
  host: '172.67.182.29',
  protocol: 'http:',
  __destroy: undefined,
  _destroy: [AsyncFunction (anonymous)],
  [Symbol(kCapture)]: false,
  [Symbol(kNeedDrain)]: false,
  [Symbol(corked)]: 0,
  [Symbol(kOutHeaders)]: [Object: null prototype] { host: [ 'host', 'example.com:443' ] }
}
URL {
  href: 'http://172.67.182.29/',
  origin: 'http://172.67.182.29',
  protocol: 'http:',
  username: '',
  password: '',
  host: '172.67.182.29',
  hostname: '172.67.182.29',
  port: '',
  pathname: '/',
  search: '',
  searchParams: URLSearchParams {},
  hash: ''
}
IncomingMessage {
  _readableState: ReadableState {
    objectMode: false,
    highWaterMark: 16384,
    buffer: BufferList { head: null, tail: null, length: 0 },
    length: 0,
    pipes: [],
    flowing: null,
    ended: true,
    endEmitted: false,
    reading: false,
    constructed: true,
    sync: true,
    needReadable: false,
    emittedReadable: true,
    readableListening: false,
    resumeScheduled: false,
    errorEmitted: false,
    emitClose: true,
    autoDestroy: true,
    destroyed: false,
    errored: null,
    closed: false,
    closeEmitted: false,
    defaultEncoding: 'utf8',
    awaitDrainWriters: null,
    multiAwaitDrain: false,
    readingMore: true,
    dataEmitted: false,
    decoder: null,
    encoding: null,
    [Symbol(kPaused)]: null
  },
  _events: [Object: null prototype] {},
  _eventsCount: 0,
  _maxListeners: undefined,
  socket: <ref *1> Socket {
    connecting: false,
    _hadError: false,
    _parent: null,
    _host: null,
    _readableState: ReadableState {
      objectMode: false,
      highWaterMark: 16384,
      buffer: BufferList { head: null, tail: null, length: 0 },
      length: 0,
      pipes: [],
      flowing: null,
      ended: false,
      endEmitted: false,
      reading: false,
      constructed: true,
      sync: false,
      needReadable: true,
      emittedReadable: false,
      readableListening: false,
      resumeScheduled: false,
      errorEmitted: false,
      emitClose: false,
      autoDestroy: true,
      destroyed: false,
      errored: null,
      closed: false,
      closeEmitted: false,
      defaultEncoding: 'utf8',
      awaitDrainWriters: null,
      multiAwaitDrain: false,
      readingMore: false,
      dataEmitted: true,
      decoder: null,
      encoding: null,
      [Symbol(kPaused)]: false
    },
    _events: [Object: null prototype] { end: [Function: onReadableStreamEnd] },
    _eventsCount: 1,
    _maxListeners: undefined,
    _writableState: WritableState {
      objectMode: false,
      highWaterMark: 16384,
      finalCalled: false,
      needDrain: false,
      ending: false,
      ended: false,
      finished: false,
      destroyed: false,
      decodeStrings: false,
      defaultEncoding: 'utf8',
      length: 0,
      writing: false,
      corked: 0,
      sync: false,
      bufferProcessing: false,
      onwrite: [Function: bound onwrite],
      writecb: null,
      writelen: 0,
      afterWriteTickInfo: null,
      buffered: [],
      bufferedIndex: 0,
      allBuffers: true,
      allNoop: true,
      pendingcb: 0,
      constructed: true,
      prefinished: false,
      errorEmitted: false,
      emitClose: false,
      autoDestroy: true,
      errored: null,
      closed: false,
      closeEmitted: false,
      [Symbol(kOnFinished)]: []
    },
    allowHalfOpen: false,
    _sockname: null,
    _pendingData: null,
    _pendingEncoding: '',
    server: null,
    _server: null,
    parser: null,
    _httpMessage: null,
    [Symbol(async_id_symbol)]: 187,
    [Symbol(kHandle)]: TCP {
      reading: true,
      onconnection: null,
      [Symbol(owner_symbol)]: [Circular *1]
    },
    [Symbol(kSetNoDelay)]: false,
    [Symbol(lastWriteQueueSize)]: 0,
    [Symbol(timeout)]: null,
    [Symbol(kBuffer)]: null,
    [Symbol(kBufferCb)]: null,
    [Symbol(kBufferGen)]: null,
    [Symbol(kCapture)]: false,
    [Symbol(kBytesRead)]: 0,
    [Symbol(kBytesWritten)]: 0,
    [Symbol(RequestTimeout)]: undefined
  },
  httpVersionMajor: 1,
  httpVersionMinor: 1,
  httpVersion: '1.1',
  complete: true,
  rawHeaders: [
    'Server',
    'cloudflare',
    'Date',
    'Sun, 19 Sep 2021 21:56:56 GMT',
    'Content-Type',
    'text/html',
    'Content-Length',
    '155',
    'Connection',
    'close',
    'CF-RAY',
    '-'
  ],
  rawTrailers: [],
  aborted: false,
  upgrade: true,
  url: '',
  method: null,
  statusCode: 400,
  statusMessage: 'Bad Request',
  client: <ref *1> Socket {
    connecting: false,
    _hadError: false,
    _parent: null,
    _host: null,
    _readableState: ReadableState {
      objectMode: false,
      highWaterMark: 16384,
      buffer: BufferList { head: null, tail: null, length: 0 },
      length: 0,
      pipes: [],
      flowing: null,
      ended: false,
      endEmitted: false,
      reading: false,
      constructed: true,
      sync: false,
      needReadable: true,
      emittedReadable: false,
      readableListening: false,
      resumeScheduled: false,
      errorEmitted: false,
      emitClose: false,
      autoDestroy: true,
      destroyed: false,
      errored: null,
      closed: false,
      closeEmitted: false,
      defaultEncoding: 'utf8',
      awaitDrainWriters: null,
      multiAwaitDrain: false,
      readingMore: false,
      dataEmitted: true,
      decoder: null,
      encoding: null,
      [Symbol(kPaused)]: false
    },
    _events: [Object: null prototype] { end: [Function: onReadableStreamEnd] },
    _eventsCount: 1,
    _maxListeners: undefined,
    _writableState: WritableState {
      objectMode: false,
      highWaterMark: 16384,
      finalCalled: false,
      needDrain: false,
      ending: false,
      ended: false,
      finished: false,
      destroyed: false,
      decodeStrings: false,
      defaultEncoding: 'utf8',
      length: 0,
      writing: false,
      corked: 0,
      sync: false,
      bufferProcessing: false,
      onwrite: [Function: bound onwrite],
      writecb: null,
      writelen: 0,
      afterWriteTickInfo: null,
      buffered: [],
      bufferedIndex: 0,
      allBuffers: true,
      allNoop: true,
      pendingcb: 0,
      constructed: true,
      prefinished: false,
      errorEmitted: false,
      emitClose: false,
      autoDestroy: true,
      errored: null,
      closed: false,
      closeEmitted: false,
      [Symbol(kOnFinished)]: []
    },
    allowHalfOpen: false,
    _sockname: null,
    _pendingData: null,
    _pendingEncoding: '',
    server: null,
    _server: null,
    parser: null,
    _httpMessage: null,
    [Symbol(async_id_symbol)]: 187,
    [Symbol(kHandle)]: TCP {
      reading: true,
      onconnection: null,
      [Symbol(owner_symbol)]: [Circular *1]
    },
    [Symbol(kSetNoDelay)]: false,
    [Symbol(lastWriteQueueSize)]: 0,
    [Symbol(timeout)]: null,
    [Symbol(kBuffer)]: null,
    [Symbol(kBufferCb)]: null,
    [Symbol(kBufferGen)]: null,
    [Symbol(kCapture)]: false,
    [Symbol(kBytesRead)]: 0,
    [Symbol(kBytesWritten)]: 0,
    [Symbol(RequestTimeout)]: undefined
  },
  _consuming: false,
  _dumped: false,
  [Symbol(kCapture)]: false,
  [Symbol(kHeaders)]: null,
  [Symbol(kHeadersCount)]: 12,
  [Symbol(kTrailers)]: null,
  [Symbol(kTrailersCount)]: 0,
  [Symbol(RequestTimeout)]: undefined
}
`

**Comment by dhavdc** [comment]
If you run the code with this proxy, you can replicate the error.
```js
gotScraping({   
    url: 'https://example.com',
    proxyUrl: 'http://172.64.75.199:80'
})
```

**Comment by dhavdc** [comment]
With axios it works perfectly fine, so the proxy works. 

```js
const HttpsProxyAgent = require('https-proxy-agent');
const agent = new HttpsProxyAgent({
    host: '172.64.75.199',
    port: '80'
})

axios({
    method: 'get',
    url: 'https://www.example.com',
    httpAgent: agent,
}).then((res) => {
    console.log(res);
}).then(() => {
    console.log('bruh')
}).catch((err) => {
    console.log(err);
})
```

Returns:

```html
  data: '<!doctype html>\n' +
    '<html>\n' +
    '<head>\n' +
    '    <title>Example Domain</title>\n' +
    '\n' +
    '    <meta charset="utf-8" />\n' +
    '    <meta http-equiv="Content-type" content="text/html; charset=utf-8" />\n' +
    '    <meta name="viewport" content="width=device-width, initial-scale=1" />\n' +
    '    <style type="text/css">\n' +
    '    body {\n' +
    '        background-color: #f0f0f2;\n' +
    '        margin: 0;\n' +
    '        padding: 0;\n' +
    '        font-family: -apple-system, system-ui, BlinkMacSystemFont, "Segoe UI", "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;\n' +
    '        \n' +
    '    }\n' +
    '    div {\n' +
    '        width: 600px;\n' +
    '        margin: 5em auto;\n' +
    '        padding: 2em;\n' +
    '        background-color: #fdfdff;\n' +
    '        border-radius: 0.5em;\n' +
    '        box-shadow: 2px 3px 7px 2px rgba(0,0,0,0.02);\n' +
    '    }\n' +
    '    a:link, a:visited {\n' +
    '        color: #38488f;\n' +
    '        text-decoration: none;\n' +
    '    }\n' +
    '    @media (max-width: 700px) {\n' +
    '        div {\n' +
    '            margin: 0 auto;\n' +
    '            width: auto;\n' +
    '        }\n' +
    '    }\n' +
    '    </style>    \n' +
    '</head>\n' +
    '\n' +
    '<body>\n' +
    '<div>\n' +
    '    <h1>Example Domain</h1>\n' +
    '    <p>This domain is for use in illustrative examples in documents. You may use this\n' +
    '    domain in literature without prior coordination or asking for permission.</p>\n' +
    '    <p><a href="https://www.iana.org/domains/example">More information...</a></p>\n' +
    '</div>\n' +
    '</body>\n' +
    '</html>\n'
}
```

**Comment by dhavdc** [comment]
Nevermind, it seems like some of the proxies I was using are just garbage so they weren't actually connecting. I retested on some other clients.

**Comment by Strajk** [comment]
For what it's worth, I was getting the same error and after pulling all my hair, I've "fixed" it by updating crawlee packages from 3.5.2 to 3.7.0, which caused updating got-scraping from v3 to v4
